lib/task/import/validate/csvValidatorResultCollection.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * CSV validation result collection class.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ */
+class CsvValidatorResultCollection
+{
+    protected $results = [];
+
+    public function __construct()
+    {
+    }
+
+    public function appendResult(CsvValidatorResult $result)
+    {
+        $this->results[] = $result;
+
+        $this->sortByFilenameStatusDescending();
+    }
+
+    public function toArray(bool $verbose = false)
+    {
+        $resultArray = [];
+
+        foreach ($this->results as $testResult) {
+            $resultArray[$testResult->getFilename()][$testResult->getClassname()] = $testResult->toArray();
+        }
+
+        return $resultArray;
+    }
+
+    public function toJson(bool $verbose = false)
+    {
+        return json_encode($this->toArray($verbose));
+    }
+
+    public function getByFilenameTestname(string $filename, string $testname)
+    {
+        foreach ($this->results as $result) {
+            if ($filename == $result->getFilename() && $testname == $result->getClassname()) {
+                return $result->toArray();
+            }
+        }
+
+        return $this->results[$filename][$testname]->toArray();
```

Comment:
I think this should be removed?

Reply:
This function is used by the PHPUnit tests to grab exactly one result array without the collection's filename/testname hierarchy.

Reply:
@sbreker I mean that the line should be removed, not the function.  I think this return duplicates is extraneous, because the preceding foreach loop does the same thing except without the risk of errors of if the $filename and $testname keys don't exist in the array.

Reply:
Oh I see now! Haha. Yes, that's shouldn't be there - leftover from a refactor.